client/components/domains/register-domain-step/index.jsx

```diff
@@ -1124,6 +1128,19 @@ class RegisterDomainStep extends Component {
 						availabilityStatus = mappable;
 					}
 
+					if (
+						[ HUNDRED_YEAR_PLAN_FLOW, HUNDRED_YEAR_DOMAIN_FLOW ].includes( this.props.flowName ) &&
+						isAvailablePremiumDomain
+					) {
+						this.removeUnavailablePremiumDomain( domain );
+						this.showSuggestionErrorMessage(
+							domain,
+							'hundred_year_domain_premium_name_restriction',
+							{}
+						);
+						return resolve( null );
```

Comment:
```suggestion
						resolve( null );
						return;
```
Do we need to `return resolve( null )` here? Or can we call `resolve( null )` and then `return`? I don't think it'll make a difference, I'm just curious.

Reply:
Well... `return` within a `Promise` is redundant - it won't have any effect as its executor passed function defines the promise's state. 
Its fulfillment values are determined by its `executor`'s parameters, `resolve` and `reject`, so `return` never changes a thing. I'll just remove it, as it can definitely cause confusion in the future. Plus it's useless 😬.